### PR TITLE
Add manual tests and test scenario for BZ-1894149

### PR DESCRIPTION
**Pull request by shubhamsg199**
**Test Results:**

```
$ pytest -v --ansible-host-pattern server --ansible-user=root  --ansible-inventory testfm/inventory tests/test_health.py -k test_positive_check_postgresql_checkpoint_segments
=============================================== test session starts =====================================================
platform linux -- Python 3.8.6, pytest-3.6.1, py-1.10.0, pluggy-0.6.0 -- /home/sganar/foreman/fm/bin/python
cachedir: .pytest_cache
ansible: 2.9.20
rootdir: /home/sganar/foreman/testfm, inifile:
plugins: ansible-2.2.3
collected 29 items / 28 deselected                                                                                                                                                                                

tests/test_health.py::test_positive_check_postgresql_checkpoint_segments PASSED                                                                                                                             [100%]

====================================== 1 passed, 28 deselected in 90.76 seconds ===============================================
```

### CODE REVIEW

File: tests/test_health.py [hunk 1]
```diff
@@ -667,20 +667,50 @@ def test_positive_check_postgresql_checkpoint_segments(ansible_module):
...
+    )
+    for result in contacted.values():
+        logger.info(result["stdout"])
+        assert (
+            "File /etc/foreman-installer/custom-hiera.yaml is not a yaml file." in result["stdout"]
+        )
+        assert "FAIL" in result["stdout"]
+        assert result["rc"] == 1
+
+    ansible_module.lineinfile(
```
Comment (Gauravtalreja1) [inline tests/test_health.py:708]: Add comment here like making it valid again

File: tests/test_health.py [hunk 2]
```diff
@@ -667,20 +667,50 @@ def test_positive_check_postgresql_checkpoint_segments(ansible_module):
...
+        line="----",
+        state="present",
+    )
+    contacted = ansible_module.command(
+        Health.check(["--label", "check-postgresql-checkpoint-segments"])
+    )
+    for result in contacted.values():
+        logger.info(result["stdout"])
+        assert (
+            "File /etc/foreman-installer/custom-hiera.yaml is not a yaml file." in result["stdout"]
```
Comment (Gauravtalreja1) [inline tests/test_health.py:703]: you can define a variable with the path of `custom-hiera` file and can use it across test

File: tests/test_health.py [hunk 3]
```diff
@@ -667,20 +667,50 @@ def test_positive_check_postgresql_checkpoint_segments(ansible_module):
         1. foreman-maintain should be installed.
 
     :steps:
-        1. Add config_entries section in /etc/foreman-installer/custom-hiera.yaml
+        1. Have an invalid /etc/foreman-installer/custom-hiera.yaml file
         2. Run foreman-maintain health check --label check-postgresql-checkpoint-segments.
```
Comment (Gauravtalreja1) [inline tests/test_health.py:671]: Also, add a step here to make custom-hiera file valid

File: tests/test_health.py [hunk 4]
```diff
@@ -667,20 +667,50 @@ def test_positive_check_postgresql_checkpoint_segments(ansible_module):
...
-        7. Remove config_entries section from /etc/foreman-installer/custom-hiera.yaml
+        7. Add checkpoint_segments parameter in /etc/foreman-installer/custom-hiera.yaml
         8. Run foreman-maintain health check --label check-postgresql-checkpoint-segments.
-        9. Assert that check-postgresql-checkpoint-segments pass.
+        9. Assert that check-postgresql-checkpoint-segments fails.
+        10. Remove config_entries section from /etc/foreman-installer/custom-hiera.yaml
+        11. Run foreman-maintain health check --label check-postgresql-checkpoint-segments.
+        12. Assert that check-postgresql-checkpoint-segments pass.
+
+    :BZ: 1894149
```
Comment (Gauravtalreja1) [inline tests/test_health.py:684]:
```suggestion
    :BZ: 1894149, 1899322
```

File: tests/test_health.py [hunk 5]
```diff
@@ -892,3 +922,30 @@ def test_positive_check_non_rh_packages(setup_custom_package, ansible_module):
         assert "walrus-5.21-1.noarch" in result["stdout"]
         assert "WARNING" in result["stdout"]
         assert result["rc"] == 78
+
+
+@stubbed
+def test_positive_available_space_postgresql(ansible_module):
```
Comment (Gauravtalreja1) [inline tests/test_health.py:928]:
```suggestion
def test_positive_available_space_postgresql12(ansible_module):
```

File: tests/test_health.py [hunk 6]
```diff
@@ -892,3 +922,30 @@ def test_positive_check_non_rh_packages(setup_custom_package, ansible_module):
...
+    """Verify Warning or Error is displayed when enough space is not
+    available under /var/opt/rh/rh-postgresql12/ after the following command.
```
Comment (Gauravtalreja1) [inline tests/test_health.py:930]:
```suggestion
    """Verify warnings when available space in /var/opt/rh/rh-postgresql12/  is less than consumed space of /var/lib/pgsql/ 
```

File: tests/test_health.py [hunk 7]
```diff
@@ -892,3 +922,30 @@ def test_positive_check_non_rh_packages(setup_custom_package, ansible_module):
...
+    :steps:
+        1. foreman-maintain health check --label available-space-for-postgresql12
+        2. Verify Warning or Error is displayed when enough space is not
+           available under /var/opt/rh/rh-postgresql12/
+
+    :BZ: 1898108, 1973363
+
+    :expectedresults: Warning or Error should be displayed after running
+                      foreman-maintain health check --label available-space-for-postgresql12
+
```
Comment (Gauravtalreja1) [inline tests/test_health.py:949]:
```suggestion
    :expectedresults: Verify warnings when available space in /var/opt/rh/rh-postgresql12/  is less than consumed space of /var/lib/pgsql/ 
```

File: tests/test_packages.py [hunk 8]
```diff
@@ -246,3 +247,24 @@ def test_positive_fm_packages_update(ansible_module, setup_packages_update):
         logger.info(result["stdout"])
         assert result["rc"] == 0
         assert "walrus-5.21-1" in result["stdout"]
+
+
+@stubbed
+def test_positive_fm_packages_sat_installer(ansible_module):
+    """Verify satellite installer is not executed after packages install/update
```
Comment (Gauravtalreja1) [inline tests/test_packages.py:254]:
```suggestion
    """Verify satellite-installer is not executed after packages install/update
```

File: tests/test_packages.py [hunk 9]
```diff
@@ -246,3 +247,24 @@ def test_positive_fm_packages_update(ansible_module, setup_packages_update):
...
+
+    :id: d73971a1-68b4-4ab2-a87c-76cc5ff80a39
+
+    :setup:
+        1. foreman-maintain should be installed.
+
+    :steps:
+        1. Run foreman-maintain packages install/update rubygem-foreman_maintain
+        2. Verify satellite installer is not executed after packages install/update
```
Comment (Gauravtalreja1) [inline tests/test_packages.py:263]:
```suggestion
        2. Verify satellite-installer is not executed after packages install/update
```

File: tests/test_packages.py [hunk 10]
```diff
@@ -246,3 +247,24 @@ def test_positive_fm_packages_update(ansible_module, setup_packages_update):
...
+
+    :BZ: 1825841
+
+    :expectedresults: satellite installer should not be executed after packages install/update
```
Comment (Gauravtalreja1) [inline tests/test_packages.py:267]:
```suggestion
    :expectedresults: satellite-installer should not be executed after packages install/update
```

File: tests/test_health.py (hunk 1 above)
Comment (shubhamsg199) [reply tests/test_health.py:710]: Noted.

File: tests/test_health.py (hunk 2 above)
Comment (shubhamsg199) [reply tests/test_health.py:705]: Okay.

**Review by Gauravtalreja1 — APPROVED** [review]

**Review by jameerpathan111 — APPROVED** [review]

**Pull request merged**